xrootd/client/session.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package client // import "go-hep.org/x/hep/xrootd/client"
+
+import (
+	"context"
+	"encoding/binary"
+	"io"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/internal/mux"
+	"go-hep.org/x/hep/xrootd/internal/xrdenc"
+	"go-hep.org/x/hep/xrootd/xrdproto"
+	"go-hep.org/x/hep/xrootd/xrdproto/signing"
+	"go-hep.org/x/hep/xrootd/xrdproto/sigver"
+)
+
+// session is a connection to the specific XRootD server
+// which allows to send requests and receive responses.
+// Concurrent requests are supported.
+// Zero value is invalid, Session should be instantiated using NewSession.
```

Comment:
`s/Session should be instantiated using NewSession./session should be instantiated using newSession./`